### PR TITLE
Allow custom buckets when building histograms for Prometheus backend

### PR DESCRIPTION
**Pull request by eschizoid**
The purpose of this PR is to add support to allow custom buckets when building histograms.

### CODE REVIEW

File: metrics/prometheus-backend/src/main/scala/sttp/client/prometheus/PrometheusBackend.scala [hunk 1]
```diff
@@ -184,7 +184,8 @@ class PrometheusListener(
   * Represents the name of a collector, together with label names and values.
   * The same labels must be always returned, and in the same order.
   */
-case class CollectorNameWithLabels(name: String, labels: List[(String, String)] = Nil) {
+case class CollectorNameWithLabels(name: String, labels: List[(String, String)] = Nil, extraBuckets: List[Double] = Nil) {
```
Comment (adamw) [inline metrics/prometheus-backend/src/main/scala/sttp/client/prometheus/PrometheusBackend.scala:187]: I'd change two things here:

1. we should make the whole list of buckets overridable. The default would be as currently (`List(.005, ...)`), but you should be able to specify your custom set
2. the name `CollectorNameWithLabels` becomes incorrect, and in fact only valid for histograms. Maybe we should have a separate `HistogramCollectorConfig` and symmetrically `CollectorConfig` for other types?

Comment (eschizoid) [reply metrics/prometheus-backend/src/main/scala/sttp/client/prometheus/PrometheusBackend.scala:187]: @adamw Thanks for reviewing this bad boy. I will take care of both comments today :)

Comment (eschizoid) [reply metrics/prometheus-backend/src/main/scala/sttp/client/prometheus/PrometheusBackend.scala]: @adamw PR comments addressed, let me know what you think

**Comment by eschizoid** [comment]
any chance to merge this bad boy?

**Comment by adamw** [comment]
Released in 2.0.8, thanks! (with slight changes :) )

**Pull request merged**